saleor/checkout/fetch.py

```diff
@@ -120,6 +128,26 @@ def update_channel_listings(self, checkout_info: "CheckoutInfo") -> None:
         )
 
 
+@dataclass(frozen=True)
+class ExternalShippingMethodInfo(DeliveryMethodBase):
```

Comment:
I would keep this class for now. `ExternalShippingMethodInfo` has some differences compared to `ShippingMethodInfo` e.g.: `update_channel_listings` and `order_key`. A single class will require multiple `if`s inside.

Reply:
I mean that we could have a single type for shipping method, which will behave in the same manner for built-in and external shipping methods. The one thing which will need to be changed in that case would be separating `models.ShippingMethod` from `ShippingMethodInfo.delivery_method`, and converting `models.ShippingMethod` into a new dataclass which will work with external and built-in shipping methods. Then there is no need to store `ShippingMethodInfo.shipping_method_channel_listings` (it is used only to fetch shipping method's price) as we could use price saved in the dataclass. 
I will not insist on this change but future PRs on changes that we're working on will have to remove this type and replace with mentioned dataclass.
Also, @maarcingebala It would be good to have your opinion here.